### PR TITLE
[ML] Fix deberta tokenizer bug caused by bug in normalizer 

### PR DESCRIPTION
**Pull request by maxhniebergall**
... which caused offsets to be negative, and caused exceptions for some very rare input combinations.

### CODE REVIEW

**Review by dan-rubinstein — APPROVED** [review]
LGTM

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/nlp/tokenizers/PrecompiledCharMapNormalizer.java [hunk 1]
```diff
@@ -194,7 +194,7 @@ Reader normalize(CharSequence str) {
                     if (charDelta < 0) {
                         // normalised form is shorter
                         int lastDiff = getLastCumulativeDiff();
-                        addOffCorrectMap(normalizedCharPos, lastDiff + charDelta);
+                        addOffCorrectMap(normalizedCharPos, lastDiff - charDelta);
```
[the remark this answers is not in the document]
Comment (maxhniebergall) [reply x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/nlp/tokenizers/PrecompiledCharMapNormalizer.java:197]: Good idea. I added a test which I confirmed fails prior to this fix, and works with this fix.

**Review by davidkyle — APPROVED** [review]
LGTM

**Comment by maxhniebergall** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.17|[](https://github.com/elastic/elasticsearch/pull/117260)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

**Pull request merged**